### PR TITLE
ARC-1699: Spicegen gRPC client binding

### PR DESCRIPTION
**Pull request by thomasrichner-oviva**
Changes:
- `spicedb-binding/` - Moved the largely borrowed `authzed-client` implementation from our permission library.
- `example/` - Basic example on how to use everything together.

We can pull out more in the future, but let's keep it managable for now. PR is already huge.

### CODE REVIEW

File: generator/out/pom.xml
[the remark this answers is not in the document]
Comment (thomasrichner-oviva) [reply generator/out/pom.xml:1]: See #3

File: example/src/test/java/com/oviva/spicegen/example/ExampleTest.java [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+                            .setObjectType(subject.kind())
+                            .setObjectId(subject.id())
+                            .build())
+                    .build())
+            .setConsistency(
+                com.authzed.api.v1.PermissionService.Consistency.newBuilder()
+                    .setAtLeastAsFresh(
+                        Core.ZedToken.newBuilder().setToken(consistencyToken).build())
+                    .build())
+            .build());
```
[the remark this answers is not in the document]
Comment (thomasrichner-oviva) [reply example/src/test/java/com/oviva/spicegen/example/ExampleTest.java:145]: Done! See #3

**Review by annadurrer-oviva — APPROVED** [review]
Looks great! Except for having a consistent naming of the mappers and its tests, everything approved!

**Comment by thomasrichner-oviva** [comment]
Thanks both! I'll address Shivan's point in the follow-up PR>

**Pull request merged**